dev_docs/tutorials/elasticsearch.mdx

```diff
@@ -0,0 +1,135 @@
+---
+id: kibDevTutorialElasticsearchClient
+slug: /kibana-dev-docs/tutorials/elasticsearch-client
+title: Access the Elasticsearch Client in a Plugin
+description: Learn how to use the Elasticsearch client in your Kibana plugin.
+date: 2025-06-19
+tags: ['kibana','onboarding', 'dev', 'elasticsearch', 'tutorials']
+---
+
+This guide covers how to obtain the Elasticsearch client in both your plugin's start lifecycle and route handler context. It explains the difference between asCurrentUser and asInternalUser, and demonstrates best practices for robust concurrent index initialization.
+
+## Getting the Elasticsearch Client
+
+### From the Plugin Start Contract
+
+In your plugin's `start` method, the `core` parameter provides access to the Elasticsearch client:
+
+```ts
+import type { CoreStart, Plugin } from '@kbn/core/server';
+
+export class MyPlugin implements Plugin {
+  public start(core: CoreStart) {
+    const esClient = core.elasticsearch.client.asCurrentUser;
+    // You can now use esClient to call Elasticsearch APIs
+  }
+}
+```
+
+- Use `asCurrentUser` to perform actions as the currently authenticated user.
+- Use `asInternalUser` for system-level operations that should bypass user permissions (see security considerations).
+
+### From a Route Handler Context
+
+When defining a server route, you can access the Elasticsearch client from the route handler's `context` parameter:
+
+```ts
+router.get(
+  { path: '/api/my_plugin/search', validate: false },
+  async (context, request, response) => {
+    const esClient = context.core.elasticsearch.client.asCurrentUser;
+    // Use esClient to call Elasticsearch APIs
+    const result = await esClient.search({
+      index: 'my-index',
+      body: { query: { match_all: {} } }
+    });
+    return response.ok({ body: result });
+  }
+);
+```
+
+- `context.core.elasticsearch.client.asCurrentUser` executes requests as the user making the HTTP request.
+- Always prefer `asCurrentUser` in route handlers to respect user permissions.
+
+## Example: Safely Initializing, Indexing, and Searching an Index
+
+When working in distributed environments (such as multiple Kibana instances), it is common for several instances to attempt to create the same index at the same time. To ensure the index is ready for ingesting and searching, all instances should:
+
+1. Attempt to create the index.
+2. If the index already exists (or creation succeeds), wait until the index status is **green** before proceeding.
```

Comment:
My only concern with preferring green over yellow is that in some cases we might never reach green... depending on how our index is configured. Maybe we can have a separate section about best practices for creating indices like using:

```
'{ "auto_expand_replicas": "0-1" }'
```

So that if there's just 1 ES node we will get to green even when the "same node" issue might have been hit for going with the default of `number_of_replicas: 1`. WDYT?

Reply:
Yeah, I agree this is sufficient no separate section needed 👍🏻